File: 0118-pascals-triangle/0118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    let res = [];

    for (let i = 0; i < numRows; i++) {
        let subList = [];

        for (let j = 0; j <= i; j++) {
            if (j === 0 || j === i) {
                subList.push(1);
            } else {
                let sum = res[i - 1][j] + res[i - 1][j - 1];
                subList.push(sum);
            }
        }

        res.push(subList);
    }

    return res;    
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        
        for(int i = 0; i < numRows; i++) {
            List<Integer> subList = new ArrayList<>();
            
            for(int j = 0; j <= i; j++) {
                if(j == 0 || j == i) {
                    subList.add(1);
                }
                else {
                    int sum = res.get(i - 1).get(j) + res.get(i - 1).get(j - 1);
                    subList.add(sum);
                }
            }
            
            res.add(subList);
        }
        
        return res;
    }
}